app/models/lobbytrack_report.rb

```diff
@@ -57,7 +72,8 @@ def self.visits_for_dates(date1, date2)
     " SET @date2 = '#{date2} 23:59:59'" \
     ' SELECT CardHolderID, DateIn, ReportField1, ReportField2, LookupField1' \
     ' FROM Jolly.dbo.logAttendance' \
-    ' WHERE DateIn between @date1 and @date2'
+    ' WHERE DateIn between @date1 and @date2' \
+    " AND CardHolderID > ''"
```

Comment:
I'm trying to understand why you added this. I suspect it is so we only get records where CardHolderID is not null, is that right?

Reply:
Yes. This will filter out all of the partially blank lines where there is no clickable ID.